internal/controller/nodeop_controller.go

```diff
@@ -585,3 +821,300 @@ func (r *NodeOpReconciler) findNodeOpsForJob(ctx context.Context, obj client.Obj
+	saName := fmt.Sprintf("%s-reboot", nodeOp.Name)
+	sa := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      saName,
+			Namespace: nodeOp.Namespace,
+		},
+	}
+	if err := controllerutil.SetControllerReference(nodeOp, sa, r.Scheme); err != nil {
+		log.Error(err, "Failed to set controller reference for service account")
+		return err
+	}
+	if err := r.Create(ctx, sa); err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			log.Error(err, "Failed to create service account")
+			return err
+		}
+	}
+
+	// Create cluster role binding for this NodeOp's service account
+	crbName := fmt.Sprintf("nodeop-reboot-%s", nodeOp.Name)
+	clusterRoleBinding := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: crbName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      saName,
+				Namespace: nodeOp.Namespace,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "nodeop-reboot",
+		},
+	}
+	if err := r.Create(ctx, clusterRoleBinding); err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			log.Error(err, "Failed to create cluster role binding")
+			return err
+		}
+	}
+
+	// Add finalizer to NodeOp if not present
+	if !controllerutil.ContainsFinalizer(nodeOp, clusterRoleBindingFinalizer) {
+		controllerutil.AddFinalizer(nodeOp, clusterRoleBindingFinalizer)
+		if err := r.Update(ctx, nodeOp); err != nil {
+			log.Error(err, "Failed to add finalizer to NodeOp")
+			return err
+		}
+	}
+
+	return nil
+}
+
+// createRebootPod creates a reboot pod that waits for a sentinel file before rebooting
+func (r *NodeOpReconciler) createRebootPod(ctx context.Context, nodeOp *kairosiov1alpha1.NodeOp, nodeName, jobBaseName string) error {
+	log := logf.FromContext(ctx)
+
+	// Ensure service account for reboot pod
+	if err := r.ensureNodeOpServiceAccount(ctx, nodeOp); err != nil {
+		log.Error(err, "Failed to ensure service account for reboot pod")
+		return err
+	}
+
+	// Get the operator image from environment variable
+	operatorImage := os.Getenv("OPERATOR_IMAGE")
+	if operatorImage == "" {
+		operatorImage = "quay.io/kairos/kairos-operator:latest"
+	}
+
+	rebootPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: fmt.Sprintf("%s-reboot-", nodeOp.Name),
+			Namespace:    nodeOp.Namespace,
+			Labels: map[string]string{
+				"kairos.io/nodeop": nodeOp.Name,
+				"kairos.io/reboot": "true",
+				"kairos.io/node":   nodeName,
+			},
+		},
+		Spec: corev1.PodSpec{
+			NodeName: nodeName,
+			HostPID:  true,
+			Containers: []corev1.Container{
+				{
+					Name:  "reboot",
+					Image: operatorImage,
+					Command: []string{
+						"/bin/sh",
+						"-c",
+						fmt.Sprintf(`
+							echo "=== Checking for existing reboot annotation ==="
+							EXISTING_ANNOTATION=$(kubectl get pod $POD_NAME --namespace $POD_NAMESPACE -o jsonpath='{.metadata.annotations.kairos\.io/reboot-state}' 2>/dev/null || echo "")
```

Comment:
In the future, instead of running a bash script in busybox, we can implement this in golang and create our own image for the reboot Pod.